### ISSUE TITLE
[@typescript-eslint/no-restricted-imports] Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by EvgenyOrekhov** [typescript-eslint/typescript-eslint#3933]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-restricted-imports": "error"
  }
}
```

OR extend the `plugin:@typescript-eslint/all` config.

```TS
export default "test";
```

`tsconfig.json`

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": ["src"]
}
```

**Expected Result**

No fatal errors.

**Actual Result**

Fatal error:

```
Oops! Something went wrong! :(

ESLint: 7.32.0

TypeError: Error while loading rule '@typescript-eslint/no-restricted-imports': Cannot convert undefined or null to object
Occurred while linting /src/test.tsx
    at hasOwnProperty (<anonymous>)
    at isObjectOfPaths (/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-restricted-imports.js:50:44)
    at isOptionsArrayOfStringOrObject (/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-restricted-imports.js:56:9)
    at getRestrictedPaths (/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-restricted-imports.js:65:9)
    at create (/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-restricted-imports.js:97:33)

```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.32.0` |
| `@typescript-eslint/parser`        | `4.32.0` |
| `TypeScript`                       | `4.4.3` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.5` |

**Comment by bradzacher** [comment]
The bug is that the extension rule assumes you pass config to the rule:
https://github.com/typescript-eslint/typescript-eslint/blob/02c6ff3c5a558f9308d7166d524156dc12e32759/packages/eslint-plugin/src/rules/no-restricted-imports.ts#L88
